### ISSUE TITLE
Centralised 0/0/0 pattern identification thread

### ISSUE THREAD

**Issue by justj0sh** [RocketMap/RocketMap#1512]
Nobody knows what triggers a 0/0/0 so I figured we should have a public repository so everyone can try to identify the pattern. Seb <3 suggested using Github Issues so here is the common thread. Also to save everyone from the spam in discord. RIP my email.

Please provide the bare minimum details:
- OS, x64/x86, 
- Scan location vs IP location
- Also your runserver parameters
- How often are you getting the errors
- Consecutive 0s?

Before you post remember to test these out:
- Remember that 0/0/0s can actually mean 0 data so test it out on a different point first
- Also try out a different account on the same IP
- Log in to your phone to test the account if possible
- If possible, try a different IP with the same account

Edit: Also, try to test a smaller st and higher sd and report back if it still gives you 0s
Edit 2: Also indicate if accounts were created with 0.35/0.43 API (make sure you accept the tos and catch a starter too)

Additional info is helpful.

Try to keep discussions out of this thread.

**Comment by HerrTapper** [comment]
Windows 10 x64
netherlands/netherlands
 -st 3 -sd 10
every time just 0.0.0, seems to be a lot of regular database cleaning going on, while i removed the database file
everytime

**Comment by eXeLe** [comment]
fresh build
Win10x64
Russia/Moscow
-st 5 -sd 15
0/0/0
checked both PTC and Google acc (checked - it works with pokemesh)

**Comment by thaiminh8xpro** [comment]
Windows 7 x64
Pleiku/Vietnam
-st 10 -sd 10
I set 6 movables, and only the central movable gets the result. Others are always 0/0/0/

**Comment by danielcha** [comment]
OS X 10.10.5 x64
Tried scanning with Dutch (The Netherlands) IPs (using VPN and own IP) for Dutch location
Using a fresh clone with the most basic command: 

`sudo python ./runserver.py -l "<location>" -st 1 -u 'user' -p 'password'`

Tried multiple accounts (that have been created over a month ago), that were successfully used for scanning in the past.

Unfortunately I can't test the accounts on a phone, but they were able to scan using PokeAlert (PokeAlert hasn't been running at the same time!)

**Comment by omgms** [comment]
I have no issue here but I just want to share my experience.

I have been scanning for more than 12 hours now, tried both hex and -ss. Return was wonderful except the despawn time. 

Some issues I found out and troubleshoot myself:

1. Scan location and Proxy/IP address should be near each other. I tried using US proxy for Singapore scanning and gotten many 0 returns. 
2. Having a short sd, big st and low account can cause captcha to happen quicker.
3. Speed limit is a problem, once an account travels above the speed limit, it needs  to rest before scanning, else it will also return 0 pokemons.

**Comment by nullpixel1** [comment]
This is because of speed limits, and IP geolocking, and IP change restriction. The map can't change or fix that

**Comment by danielcha** [comment]
Can you explain the IP change restriction? Every account can only change IPs every so often (x times per hour)?

**Comment by pogarek** [comment]
* OS, x64
* Scan location vs IP location :      the same (Warsaw, poland) 
* Also your runserver parameters :        -sd 15 --step-limit 2 --location --port 8000 --db xx.db  --gmaps-key --auth-service ptc  . 1 account only (for testing 0.0.0.0 issue) 
* How often are you getting the errors:       since today I've started testing the latest release. I've turned off the map two months ago

**Comment by faketits** [comment]
fresh install from repo
linux 14.04
Israel
-st 2 -sd 15
i'm currently running only 1 worker.
tried 3 different IP address tried to move the worker around in a reasonable distance... nothing.

**Comment by shadi1989** [comment]
Windows 10/x64
netherlands/netherlands
SD 15
19instances
- 10 accounts per instance
- ST 10 per instance

from the 10 accounts in 1 instance, about half of them work normally, other half is getting 0/0/0.

**Comment by faketits** [comment]
pip install -r requirements.txt --upgrade
should fix your issue

**Comment by nullpixel1** [comment]
How?
There were no commits to the repos that that pulls from @faketits 

@FrostTheFox there are way to many repros here: can you explain some of the factors to this are not controllable, such as IP geolocking & IP changes. In theory, those two things should only affect big maps

**Comment by joekeating87** [comment]
Ubuntu Server 16.0.4.1 x64 

Location IP: Thailand 
Location Scanned: Thailand

Running 1 instance with 1 worker to test.

Fresh account (created 7 weeks ago but never logged in) 

Scan delay: 20
Step size: 3

Everything seems to be working well with 1 account 1 instance. 

I tried old accounts that had previously encountered captcha and they would return 0/0/0 even on this low setting.

I tried my 65 instances at -st 5 -sd 20  with 195 fresh accounts and it was terrible. 1-2 Pokémon found and the rest was 0/0/0.

**Comment by gatomiaurex** [comment]
Couldn't the 0/0/0 results be because of the account triggering the captcha? I ran two instances

1.- 10 accounts previously used with pokemesh with solved captcha = 100+ finds per account
2.- 10 accounts that haven't been logued in since the API change, therefore no solved captcha = tons of 0/0/0 results

I will try to solve the captcha of the second instance to see if it starts finding pokemons, will report back.

**Comment by cfsterpka** [comment]
Ubuntu 16.04 x86-64
IP/Scan Location: US
 params: -gi -st 5 -sd 30
Always getting 0/0/0, have only tried PTC accounts.  None of my old accounts work, have tried re-solving the captcha but this doesn't seem to change anything.  Created a two new PTC accounts and neither have prompted for captchas, both are getting 0/0/0.

**Comment by Mattyx89** [comment]
Does anyone know a program on windows to create new PTC account?

**Comment by tomballgithub** [comment]
I've been scanning 7 hours with 25 previously used accounts and -st 10 -sd 10 -fl
- All accounts had TOS accepted but not an initial pokemon caught
- I loaded all accounts into Pokealarm and did the captchas before starting

0 captchas or banned accounts

**Comment by m3g4mInD** [comment]
@tomballgithub ok i gonna try that if work

**Comment by tomballgithub** [comment]
I'd like to do < 25 accounts but have not tried it yet.  Has anyone else tried less with -st 10 -sd 10?

**Comment by jorgitoh** [comment]
@tomballgithub Pokealarm or pokealert? IOS or Android?

**Comment by tomballgithub** [comment]
@jorgitoh Pokealert Android (I fixed it above)

**Comment by dwenzier** [comment]
Windows 10 x64
netherlands/netherlands
-st 5 -sd 10 -lp 3
Always 0.0.0 on all accounts
Checked them with Pokealert, all were ok.

After updating pip, as @faketits suggested, I started getting results!
Just 1 account still getting 0.0.0.

**Comment by tomballgithub** [comment]
I've been hex scanning since yesterday with -st 10 -sd 10 -fl
- All accounts had TOS accepted but not an initial pokemon caught
- I loaded all accounts into Pokealert Android and did the captchas before starting
- Not doing encounters or gym info
- Using proxy via Private Internet Access server that is in same state as I am scanning
- Windows Home Server 2011 x64

**Hex Scan 1:**
- 25 accounts
- 21 hours (started 1pm CST 11/6)
- 0 captchas or banned accounts

**Hex Scan 2:**
- 20 accounts
- 12 hours (started 10pm CST 11/6)
- 0 captchas or banned accounts

**Hex Scan 3:**
- 20 accounts
- 12 hours (started 10pm CST 11/6)
- 0 captchas or banned accounts

**Comment by cfsterpka** [comment]
Hmm... my accounts seem to be working fine with PokeAlert, however even after updating pip and doing a clean install of PokemonGo-Map I still am getting 0/0/0!

**Comment by NeverEnoughTime** [comment]
Windows 7 64bit
South Dakota, USA / same
-sl 4, -sd 20 for my intial testing
No errors at this time, I do get 0's but I those are just the area, it's not densly populated.

Only using 3 of my 10 accounts at the moment, all from 0.35 but tested and captcha solved with pokealert yesterday.

Only running a single instance.

**Comment by forcewill** [comment]
Raspbian 32Bit
sd and st default 
Using google accounts
Running two instances one with 1 account the other with two

**2016-11-08 16:22:34,219 [ search-worker-0][   auth_google][    INFO] Forced request of Google Access Token!
2016-11-08 16:22:34,575 [ search-worker-0][        pgoapi][   ERROR] Request for new Access Token failed! Logged out...
2016-11-08 16:22:34,576 [ search-worker-0][        search][ WARNING] Exception while downloading map: Request for new Access Token failed! Logged out...
**

After this i get cool down is this related to this issue ? if not im sry

**Comment by jsherin** [comment]
I've found an -sd of 30 prevents 0 pokemon. Even 20 causes me to have multiple 0 results in a row sometimes.

**Comment by chromer90** [comment]
Windows 10 64 Bit
Locations are the same as IP
using -st 3 and -sd 30
No errors
Only getting 0/0/0


Currently I am using 2 different machines to run 2 separate servers for different reasons.

Both are setup the same though. Did the latest pull and updated everything and installed everything.

both are using only 1 account to scan a small area. Both accounts have logged into PoGo and completed captcha.

However all I get is 0/0/0

We know it is densely populated because we are getting tons of spawns on our phones.

Any ideas? One machine gets a dynamic IP so it doesn't appear that that is the issue. The other one has the same IP for months and I tried a VPN and they all gave me the banned IP message.

**Comment by forcewill** [comment]
@everyone just wondering to RPI users here,
which **libencrypt-linux-arm-32.so** and **libniantichash-linux-arm-32.so** are you guys using ?

Anyone tried the new: https://github.com/pokecrypt/pokecrypt-pgoapi/pull/1/commits/df16782e8606eddf6939c515524f036a86699a95#diff-006fa9871939712720456d941377b3b6

**Comment by popcornmix** [comment]
I've checked and they are identical to the versions I'm successfully running, so I say they are good.

**Comment by cfsterpka** [comment]
Hmm, does the arch matter?  I'm running the x86-64bit version in Ubuntu.  Possibly related: I noticed PokeAlert runs fine on my Android device, but detects nothing when run through Genymotion on my x86-64 Ubuntu.

**Comment by cfsterpka** [comment]
Ha!  The latest update fixed it!

**Comment by bcintor** [comment]
Despite starting with a fresh install had to do what @faketits recommended to eliminate the 0/0/0 results.

**Comment by cfsterpka** [comment]
+1 for @faketits recommendation.  I just set up 3 new servers and needed to run "pip install -r requirements.txt --upgrade" to get them to work.  Seems a bit odd that this is required even in new installs.  Maybe this should be added to the documentation?

**Comment by skaynl** [comment]
this behavior started for me yesterday on november 22, 2016. Tried clean install, checked PokeAlert 4.0.4-7 in bluestack for al the account (all good), Ran updates/upgrades (requirement.txt) and what ever I could think off. Did a "pip freeze | xargs pip uninstall -y" as the wiki recommends. But nothing helped. I do the reccomended updates on my Ubuntu 16 machine daily, did I update something that broke pokemap??

**Comment by skaynl** [comment]
OK DON'T do that pip uninstall thing, after that my letsencrypt (to renew certificates on my server) was broken!

**Comment by cfsterpka** [comment]
Have you logged into your accounts on a device to see if there's a captcha?  The auto-captcha doesn't seem to work 100% of the time and I have to log in to all of my accounts usually about once a day to manually solve the captchas.

**Comment by skaynl** [comment]
well I did check them with PokeAlert on a bluestack installation, but I'll try one to make sure.

**Comment by skaynl** [comment]
Guess PokeAlert on bluestack didn't work so well.... first account I tried got a captcha screen

**Comment by sebastienvercammen** [comment]
Closing this issue because it's very old and no longer relevant to the current codebase. If the issue still persists for you today, comment on this issue to reopen it with new details (logs, screenshots).
